internal/build/container_ops.go

```diff
@@ -207,92 +207,60 @@ func findMount(info types.ContainerJSON, dst string) (types.MountPoint, error) {
 	return types.MountPoint{}, fmt.Errorf("no matching mount found for %s", dst)
 }
 
-// WriteProjectMetadata
-func WriteProjectMetadata(p string, metadata platform.ProjectMetadata, os string) ContainerOperation {
-	return func(ctrClient DockerClient, ctx context.Context, containerID string, stdout, stderr io.Writer) error {
-		buf := &bytes.Buffer{}
-		err := toml.NewEncoder(buf).Encode(metadata)
-		if err != nil {
-			return errors.Wrap(err, "marshaling project metadata")
-		}
+func writeToml(ctrClient DockerClient, ctx context.Context, data interface{}, dstPath string, containerID string, os string, stdout, stderr io.Writer) error {
+	buf := &bytes.Buffer{}
+	err := toml.NewEncoder(buf).Encode(data)
+	if err != nil {
+		return errors.Wrap(err, "marshaling data")
+	}
 
-		tarBuilder := archive.TarBuilder{}
+	tarBuilder := archive.TarBuilder{}
 
-		tarPath := p
-		if os == "windows" {
-			tarPath = paths.WindowsToSlash(p)
-		}
+	tarPath := dstPath
+	if os == "windows" {
+		tarPath = paths.WindowsToSlash(dstPath)
+	}
 
-		tarBuilder.AddFile(tarPath, 0755, archive.NormalizedDateTime, buf.Bytes())
-		reader := tarBuilder.Reader(archive.DefaultTarWriterFactory())
-		defer reader.Close()
+	tarBuilder.AddFile(tarPath, 0755, archive.NormalizedDateTime, buf.Bytes())
+	reader := tarBuilder.Reader(archive.DefaultTarWriterFactory())
+	defer reader.Close()
 
-		if os == "windows" {
-			dirName := paths.WindowsDir(p)
-			return copyDirWindows(ctx, ctrClient, containerID, reader, dirName, stdout, stderr)
-		}
+	if os == "windows" {
+		dirName := paths.WindowsDir(dstPath)
+		return copyDirWindows(ctx, ctrClient, containerID, reader, dirName, stdout, stderr)
+	}
+
+	return ctrClient.CopyToContainer(ctx, containerID, "/", reader, types.CopyToContainerOptions{})
+}
 
-		return ctrClient.CopyToContainer(ctx, containerID, "/", reader, types.CopyToContainerOptions{})
+// WriteToml writes a `data.toml` for test only.
+func WriteToml(dstPath string, data interface{}, os string) ContainerOperation {
```

Comment:
I think rather than introduce a new public API function only for the tests, it would be better to test one of the existing public functions